docs/creating_new_plugins.md

```diff
@@ -0,0 +1,113 @@
+
+## Write Dart Code
+
+1. Implement your dart code in `{yourPluginName}.dart` file. You will want to implement the public APIs here and make sure the signature matches as closely as possible to the native API. You will simply invoke the native method using the `_channel.invokeMethod` API. 
+
+2. Write Dart tests in your `test/` directory. The dart tests should simply make sure that the correct values are being passed to and from the bridge, and that the correct methods are being invoked.
+
+
+
+## Update Sample App (optional)
+
+If you want this plugin to be included in the sample app, then you need to do the following:
+
+1. Navigate to the monorepo's `aepsdk_flutter/example/pubspec.yml` file, and include your plugin as a path dependency (This is for active development. Once your plugin is published, you can change it to point to the published plugin instead.). To include your plugin via path add the following to the `pubspec.yml` under the `dev_dependencies:` section: 
+
+```
+...
+
+dev_dependencies:
+
+your_plugin_name:
+    path: ../plugins/your_plugin_name
+
+...
+
+```
+2. Next you will update the example app dart code under `example/lib/main.dart`. First, import your package at the top:
+`import 'package:flutter_{yourPluginName}/flutter_{yourPluginName}.dart'`. 
+
+3. How the example app will use your plugin is up to you, but simply add integrations with your public APIs to the UI.
+
+4. Next, you will update the example app native code. 
+
+    a. For iOS, update the AppDelegate to register your extension. If your extension requires native code support to work, add it now.
+
+    b. For Android, update the MyApplication.java to register the new extension. If your extension requires native code support to work, add it now.
+
+5. To make sure the ios project is configured correctly, run "flutter pub get" from the "example" directory. Then run a "pod install" from the ios directory. Now open the Runner.xcworkspace and make sure it builds correctly.
+
+6. To make sure the Android project is configured correctly, open the android directory in Android Studio and make sure the automated gradle scripts work properly. 
+
+## Testing:
+
+1. Now, you are ready to test your plugin. Make sure that you have run `flutter pub get` from the example directory, and then a `pod install` from the ios directory.
+
+2. Set up a launch property and add the app id to the native code for both android and ios. 
+
+3. Set up an Assurance session. Note that for android the host url is: "aepsdkflutter://adobe" and for iOS it is: "aepsdkflutter://host"
```

Comment:
Is the example app configured with the following deeplink url?

Reply:
Yep